src/parser/node.js

```diff
@@ -1,48 +1,68 @@
 import Parser from "./index";
 import { SourceLocation, type Position } from "../util/location";
+import * as N from "../types";
+import type { NodeBase } from "../types";
 
 // Start an AST node, attaching a start offset.
 
-const pp = Parser.prototype;
 const commentKeys = ["leadingComments", "trailingComments", "innerComments"];
 
-class Node {
+class Node implements NodeBase {
+  start: number;
+  end: number;
+  loc: SourceLocation;
+  range: [number, number];
+  leadingComments: ?Array<N.Comment>;
+  trailingComments: ?Array<N.Comment>;
+  innerComments: ?Array<N.Comment>;
+  extra: { [key: string]: any };
+
   constructor(parser?: Parser, pos?: number, loc?: Position) {
+    // $FlowIgnore
     this.type = "";
+    // $FlowNonNull
     this.start = pos;
```

Comment:
`new Node;` is called in one place, but I just added a `$FlowFixMe` comment at that place and no longer need hacks here. Thanks!